diff --git a/js/src/jsapi.h b/js/src/jsapi.h
index 2940624..2370457 100644
--- a/js/src/jsapi.h
+++ b/js/src/jsapi.h
@@ -2399,7 +2399,7 @@ JS_FreezeObject(JSContext* cx, JS::Handle<JSObject*> obj);
 
 /*** Property descriptors ************************************************************************/
 
-struct JSPropertyDescriptor : public JS::Traceable {
+struct JS_PUBLIC_API(JSPropertyDescriptor) : public JS::Traceable {
     JSObject* obj;
     unsigned attrs;
     JSGetterOp getter;
